tests/unit/keyinput/test_basekeyparser.py

```diff
@@ -205,12 +239,39 @@ def test_ambiguous_keychain(self, qapp, fake_keyevent_factory, config_stub,
         assert not timer.isActive()
         assert self.kp._keystring == ''
 
+    def test_ambiguous_keychain_no_timeout(self, fake_keyevent_factory,
+                                           config_stub, monkeypatch):
+        """Test ambiguous keychain with timeout equal to 0."""
+        config_stub.data = CONFIG_NO_TIMEOUT
+        monkeypatch.setattr('qutebrowser.keyinput.basekeyparser.config',
+                            config_stub)
```

Comment:
I know you just copied this from my test - but I think this `monkeypatch` line is actually unnecessary - can you try without?